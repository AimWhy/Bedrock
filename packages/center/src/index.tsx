import { mergeBreakpoints } from "@bedrock-layout/spacing-constants";
import PropTypes from "prop-types";
import styled from "styled-components";

export interface CenterProps {
  maxWidth?: number | string;
  centerText?: boolean;
  centerChildren?: boolean;
}

function getSafeMaxWidth(
  breakPoints: Record<string, unknown>,
  maxWidth?: number | string
) {
  if (typeof maxWidth === "string") {
    return maxWidth;
  }

  return typeof maxWidth === "number"
    ? `${maxWidth}px`
    : mergeBreakpoints(breakPoints).medium + "px";
}

export const Center = styled.div.attrs<CenterProps>(() => {
  return {
    "data-bedrock-layout-center": "",
  };
})<CenterProps>`
  @property --maxWidth {
    syntax: "<length>";
    inherits: false;
    initial-value: 1023px;
  }
  --maxWidth: ${({ maxWidth, theme: { breakPoints } }) =>
    getSafeMaxWidth(breakPoints, maxWidth)};

  box-sizing: content-box;

  margin-inline-start: auto;
  margin-inline-end: auto;
  margin-inline: auto;

  max-inline-size: 1023px;

  @supports (
    max-inline-size:
      ${({ maxWidth, theme: { breakPoints } }) =>
        getSafeMaxWidth(breakPoints, maxWidth)}
  ) {
    max-inline-size: var(--maxWidth, 1023px);
  }

  ${(props) => (props.centerText ? "text-align: center;" : "")}

  ${(props) =>
    props.centerChildren
      ? `
  display: flex;
  flex-direction: column;
  align-items: center;
  `
      : ""}
`;

Center.displayName = "Center";

Center.propTypes = {
  maxWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  centerText: PropTypes.bool,
};
