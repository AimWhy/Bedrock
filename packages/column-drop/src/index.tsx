import {
  CSSLength,
  Gutter,
  SizesOptions,
  getSafeGutter,
  getSizeValue,
  useTheme,
} from "@bedrock-layout/spacing-constants";
import { forwardRefWithAs } from "@bedrock-layout/type-utils";
import React, { CSSProperties } from "react";

type MinItemWidth = CSSLength | number | SizesOptions;

export interface ColumnDropProps {
  gutter?: Gutter;
  minItemWidth?: MinItemWidth;
  noStretchedColumns?: boolean;
}

function getSafeMinItemWidth(
  theme: {
    space?: { [key: string]: string };
    sizes?: { [key: string]: string };
  },
  minItemWidth?: MinItemWidth,
) {
  return getSizeValue(theme, minItemWidth);
}

export const ColumnDrop = forwardRefWithAs<"div", ColumnDropProps>(
  (
    { as, gutter, style, minItemWidth, noStretchedColumns = false, ...props },
    ref,
  ) => {
    const theme = useTheme();
    const maybeGutter = getSafeGutter(theme, gutter);

    const attributeValue =
      noStretchedColumns === true ? "no-stretched-columns" : "";

    const safeMinItemWidth = getSafeMinItemWidth(theme, minItemWidth);
    const safeStyle = style ?? {};

    const Component = as ?? "div";
    return (
      <Component
        ref={ref}
        data-bedrock-column-drop={attributeValue}
        style={
          {
            ...safeStyle,
            "--gutter": maybeGutter,
            "--minItemWidth": safeMinItemWidth,
          } as CSSProperties
        }
        {...props}
      />
    );
  },
);
ColumnDrop.displayName = "ColumnDrop";
