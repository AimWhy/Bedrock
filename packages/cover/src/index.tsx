import {
  CSSLength,
  Gutter,
  SizesOptions,
  getSafeGutter,
  getSizeValue,
  useTheme,
} from "@bedrock-layout/spacing-constants";
import { forwardRefWithAs } from "@bedrock-layout/type-utils";
import React, { CSSProperties } from "react";

/**
 * `minHeight` can be a CSSLength, a number, or a key of the theme's sizes object
 */
export type MinHeight = CSSLength | number | SizesOptions;

/**
 * Props for the Cover component.
 */
export type CoverProps = {
  /**
   * Slot to be placed at the top of the cover component, above the centered content.
   */
  top?: React.ReactNode;
  /**
   * Slot to be placed at the bottom of the cover component, below the centered content.
   */
  bottom?: React.ReactNode;
  /**
   * Sets space between each element.
   * @deprecated Use `gap` instead.
   */
  gutter?: Gutter;
  /**
   * Sets space between each element.
   */
  gap?: Gutter;
  /**
   * Sets the minimum height of the cover component.
   * `minHeight` can be a CSSLength, a number, or a key of the theme's sizes object
   * @default "100vh"
   */
  minHeight?: MinHeight;
  /**
   * Sets the content to stretch to the full height of the cover component minus the top and bottom slots.
   */
  stretchContent?: boolean;
};

/**
 * The `Cover` component is designed to vertically cover a predefined area, `100vh` by default, and vertically center its children.
 * You can also conditionally render a top and/or bottom slot as well.
 */
export const Cover = forwardRefWithAs<"div", CoverProps>(function Cover(
  {
    as: Component = "div",
    children,
    gap,
    gutter,
    top,
    bottom,
    minHeight,
    style = {},
    //TODO: Change from boolean to string union of "stretch-content" | "default"
    stretchContent,
    ...props
  },
  ref,
) {
  const theme = useTheme();
  const maybeGutter = getSafeGutter(theme, gap ?? gutter);
  const maybeMinHeight = getSizeValue(theme, minHeight);

  const attributeVal = stretchContent === true ? "stretch-content" : "";

  return (
    <Component
      ref={ref}
      data-bedrock-cover={attributeVal}
      style={
        {
          "--gutter": maybeGutter,
          "--minHeight": maybeMinHeight,
          ...style,
        } as CSSProperties
      }
      {...props}
    >
      {top && <div data-bedrock-cover-top="">{top}</div>}
      <div data-bedrock-cover-centered="">{children}</div>
      {bottom && <div data-bedrock-cover-bottom="">{bottom}</div>}
    </Component>
  );
});
