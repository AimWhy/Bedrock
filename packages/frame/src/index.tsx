import { forwardRefWithAs } from "@bedrock-layout/type-utils";
import React, { CSSProperties } from "react";
type RatioString = `${number}/${number}` | `${number} / ${number}`;

type Ratio = [number, number] | RatioString;
export interface FrameProps {
  ratio?: Ratio;
  position?: string;
}

function checkIsRatio(ratio: unknown): ratio is Ratio {
  const isCorrectArray =
    Array.isArray(ratio) && ratio.length === 2 && ratio.every(Number.isFinite);
  return (
    isCorrectArray ||
    (typeof ratio === "string" &&
      /^\d{1,1000} {0,1}\/ {0,1}\d{1,1000}$/.test(ratio))
  );
}

function getRatioString(ratio: Ratio): RatioString {
  return Array.isArray(ratio) ? (ratio.join("/") as RatioString) : ratio;
}

function getSafeRatio(ratio: unknown): RatioString | undefined {
  const isRatio = checkIsRatio(ratio);

  return isRatio ? getRatioString(ratio) : undefined;
}

export const Frame = forwardRefWithAs<"div", FrameProps>(
  ({ as, ratio, style, position, ...props }, ref) => {
    const safeRatio = getSafeRatio(ratio);
    const safeStyle = style ?? {};

    const Component = as ?? "div";

    return (
      <Component
        data-bedrock-frame
        ref={ref}
        style={
          {
            ...safeStyle,
            "--ratio": safeRatio,
            "--position": position,
          } as CSSProperties
        }
        {...props}
      />
    );
  }
);

Frame.displayName = "Frame";
