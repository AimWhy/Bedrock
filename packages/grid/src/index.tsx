import {
  SpacingOptions,
  getSpacingValue,
} from "@bedrock-layout/spacing-constants";
import PropTypes from "prop-types";
import styled from "styled-components";

const SMALL_BREAKPOINT = "639px";

export interface GridProps {
  gutter: keyof SpacingOptions;
  minItemWidth?: number | string;
}

function getSafeMinItemWidth(minItemWidth?: number | string) {
  if (typeof minItemWidth === "string") {
    return minItemWidth;
  }

  return typeof minItemWidth === "number"
    ? `${minItemWidth}px`
    : SMALL_BREAKPOINT;
}

export const Grid = styled.div.attrs<GridProps>(() => {
  return {
    "data-bedrock-layout-grid": "",
  };
})<GridProps>`
  box-sizing: border-box;
  @property --minItemWidth {
    syntax: "<length-percentage>";
    inherits: false;
    initial-value: ${SMALL_BREAKPOINT};
  }

  --gutter: ${({ gutter, theme }) => getSpacingValue(theme, gutter) ?? "0px"};
  --minItemWidth: ${({ minItemWidth }) => getSafeMinItemWidth(minItemWidth)};

  display: grid;
  gap: var(--gutter);

  grid-template-columns: repeat(
    auto-fit,
    minmax(min(${SMALL_BREAKPOINT}, 100%), 1fr)
  );

  @supports (
    width: ${({ minItemWidth }) => getSafeMinItemWidth(minItemWidth)}
  ) {
    grid-template-columns: repeat(
      auto-fit,
      minmax(min(var(--minItemWidth), 100%), 1fr)
    );
  }
`;

Grid.displayName = "Grid";

Grid.propTypes = {
  gutter: PropTypes.string.isRequired as React.Validator<keyof SpacingOptions>,
  minItemWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};
