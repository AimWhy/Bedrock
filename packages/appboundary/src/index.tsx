import { Center } from "@bedrock-layout/center";
import { mergeBreakpoints } from "@bedrock-layout/spacing-constants";
import React from "react";
import styled, { ThemeContext } from "styled-components";

const safeTheme = { breakPoints: {} };

export const AppBoundary = styled.div.attrs((props) => {
  const { breakPoints } = React.useContext(ThemeContext) ?? safeTheme;
  const maxWidth = mergeBreakpoints(breakPoints).xxlarge;
  return {
    "data-bedrock-layout-appboundary": "",
    children: <Center maxWidth={maxWidth}>{props.children}</Center>,
  };
})`
  padding: 0;
  max-width: 100%;
  overflow: hidden;
  height: 100%;
`;

AppBoundary.displayName = "AppBoundary";
