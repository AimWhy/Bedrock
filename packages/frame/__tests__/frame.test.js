import React from "react";
import { create } from "react-test-renderer";

import { Frame } from "../src";

describe("Frame", () => {
  describe("correct usage", () => {
    test("Frame is not null", () => {
      expect(Frame).toBeTruthy();
    });

    it("renders with ratio as an array", () => {
      const frame = create(
        <Frame ratio={[16, 9]}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });

    it("renders with ratio as a string", () => {
      const frame = create(
        <Frame ratio="16/9">
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });

    it("renders without ratio", () => {
      const frame = create(
        <Frame>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });

    it("renders custom position", () => {
      const frame = create(
        <Frame ratio={[16, 9]} position="top left">
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });
  });

  describe("incorrect usage", () => {
    beforeEach(() => {
      jest.spyOn(console, "error");
      console.error.mockImplementation(() => undefined);
    });
    afterEach(() => {
      console.error.mockRestore();
    });

    it("renders default position with incorrect value", () => {
      expect(console.error).not.toBeCalled();

      const errorStack = create(
        <Frame ratio={[16, 9]} position={true}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders without ratio if ratio is not correct type", () => {
      expect(console.error).not.toBeCalled();

      const errorStack = create(
        <Frame ratio={{ 0: 16, 1: 9 }}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders without ratio with error if array of length <1 provided", () => {
      expect(console.error).not.toBeCalled();

      const errorStack = create(
        <Frame ratio={[16]}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders without ratio with error if array of length >2 provided", () => {
      expect(console.error).not.toBeCalled();

      const errorStack = create(
        <Frame ratio={[16, 9, 8, 6]}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders without ratio with error if array of not numbers provided", () => {
      expect(console.error).not.toBeCalled();

      const errorStack = create(
        <Frame ratio={["16", "9"]}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders without ratio with error if ratio string is not correct format", () => {
      expect(console.error).not.toBeCalled();

      const errorStack = create(
        <Frame ratio="16:9">
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });
  });
});
