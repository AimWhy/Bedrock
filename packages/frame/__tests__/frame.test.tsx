import React from "react";
import { create } from "react-test-renderer";
import { describe, expect, it, test } from "vitest";

import { Frame } from "../src";

describe("Frame", () => {
  describe("correct usage", () => {
    test("Frame is not null", () => {
      expect(Frame).toBeTruthy();
    });

    it("renders with ratio as an array", () => {
      const frame = create(
        <Frame ratio={[16, 9]}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });

    it("renders with ratio as a string", () => {
      const frame = create(
        <Frame ratio="16/9">
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });

    it("renders with ratio as a string with spaces", () => {
      const frame = create(
        <Frame ratio="16 / 9">
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });

    it("renders with ratio string using a colon", () => {
      const frame = create(
        <Frame ratio="16:9">
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });

    it("renders with ratio string using a colon and with spaces", () => {
      const frame = create(
        <Frame ratio="16 : 9">
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });

    it("renders without ratio", () => {
      const frame = create(
        <Frame>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });

    it("renders custom position", () => {
      const frame = create(
        <Frame ratio={[16, 9]} position="top left">
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });
  });

  describe("incorrect usage", () => {
    it("renders default position with incorrect value", () => {
      const errorStack = create(
        // @ts-expect-error
        <Frame ratio={[16, 9]} position={true}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders without ratio if ratio is not correct type", () => {
      const errorStack = create(
        // @ts-expect-error
        <Frame ratio={{ 0: 16, 1: 9 }}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders without ratio with error if array of length <1 provided", () => {
      const errorStack = create(
        // @ts-expect-error
        <Frame ratio={[16]}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders without ratio with error if array of length >2 provided", () => {
      const errorStack = create(
        // @ts-expect-error
        <Frame ratio={[16, 9, 8, 6]}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders without ratio with error if array of not numbers provided", () => {
      const errorStack = create(
        // @ts-expect-error
        <Frame ratio={["16", "9"]}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders without ratio with error if ratio string is not correct format", () => {
      const errorStack = create(
        <Frame ratio="16:9">
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(errorStack.toJSON()).toMatchSnapshot();
    });
  });
});
