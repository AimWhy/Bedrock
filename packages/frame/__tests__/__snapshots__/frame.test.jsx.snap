// Vitest Snapshot v1

exports[`Frame > correct usage > renders custom position 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  position: relative;
  overflow: hidden;
}

.c0[style*="--ratio"] {
  aspect-ratio: var(--ratio);
}

.c0 > * {
  position: absolute;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :is(img,video) {
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  size: 100%;
  object-fit: cover;
  object-position: top left;
}

<div
  className="c0"
  data-bedrock-frame=""
  style={
    {
      "--ratio": "16/9",
    }
  }
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame > correct usage > renders with ratio as a string 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  position: relative;
  overflow: hidden;
}

.c0[style*="--ratio"] {
  aspect-ratio: var(--ratio);
}

.c0 > * {
  position: absolute;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :is(img,video) {
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  size: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
  data-bedrock-frame=""
  style={
    {
      "--ratio": "16/9",
    }
  }
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame > correct usage > renders with ratio as a string with spaces 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  position: relative;
  overflow: hidden;
}

.c0[style*="--ratio"] {
  aspect-ratio: var(--ratio);
}

.c0 > * {
  position: absolute;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :is(img,video) {
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  size: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
  data-bedrock-frame=""
  style={
    {
      "--ratio": "16 / 9",
    }
  }
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame > correct usage > renders with ratio as an array 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  position: relative;
  overflow: hidden;
}

.c0[style*="--ratio"] {
  aspect-ratio: var(--ratio);
}

.c0 > * {
  position: absolute;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :is(img,video) {
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  size: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
  data-bedrock-frame=""
  style={
    {
      "--ratio": "16/9",
    }
  }
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame > correct usage > renders without ratio 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  position: relative;
  overflow: hidden;
}

.c0[style*="--ratio"] {
  aspect-ratio: var(--ratio);
}

.c0 > * {
  position: absolute;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :is(img,video) {
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  size: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
  data-bedrock-frame=""
  style={
    {
      "--ratio": undefined,
    }
  }
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame > incorrect usage > renders default position with incorrect value 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  position: relative;
  overflow: hidden;
}

.c0[style*="--ratio"] {
  aspect-ratio: var(--ratio);
}

.c0 > * {
  position: absolute;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :is(img,video) {
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  size: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
  data-bedrock-frame=""
  style={
    {
      "--ratio": "16/9",
    }
  }
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame > incorrect usage > renders without ratio if ratio is not correct type 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  position: relative;
  overflow: hidden;
}

.c0[style*="--ratio"] {
  aspect-ratio: var(--ratio);
}

.c0 > * {
  position: absolute;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :is(img,video) {
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  size: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
  data-bedrock-frame=""
  style={
    {
      "--ratio": undefined,
    }
  }
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame > incorrect usage > renders without ratio with error if array of length <1 provided 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  position: relative;
  overflow: hidden;
}

.c0[style*="--ratio"] {
  aspect-ratio: var(--ratio);
}

.c0 > * {
  position: absolute;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :is(img,video) {
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  size: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
  data-bedrock-frame=""
  style={
    {
      "--ratio": undefined,
    }
  }
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame > incorrect usage > renders without ratio with error if array of length >2 provided 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  position: relative;
  overflow: hidden;
}

.c0[style*="--ratio"] {
  aspect-ratio: var(--ratio);
}

.c0 > * {
  position: absolute;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :is(img,video) {
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  size: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
  data-bedrock-frame=""
  style={
    {
      "--ratio": undefined,
    }
  }
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame > incorrect usage > renders without ratio with error if array of not numbers provided 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  position: relative;
  overflow: hidden;
}

.c0[style*="--ratio"] {
  aspect-ratio: var(--ratio);
}

.c0 > * {
  position: absolute;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :is(img,video) {
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  size: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
  data-bedrock-frame=""
  style={
    {
      "--ratio": undefined,
    }
  }
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame > incorrect usage > renders without ratio with error if ratio string is not correct format 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  position: relative;
  overflow: hidden;
}

.c0[style*="--ratio"] {
  aspect-ratio: var(--ratio);
}

.c0 > * {
  position: absolute;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :is(img,video) {
  inline-size: 100%;
  -webkit-block-size: 100%;
  -ms-flex-block-size: 100%;
  block-size: 100%;
  size: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
  data-bedrock-frame=""
  style={
    {
      "--ratio": undefined,
    }
  }
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;
